### PR TITLE
LPS-168539 Use BundleUtil.getBundle method instead of repeating its code again and again

### PR DESCRIPTION
**Pull request by jorgediaz-lr**
https://issues.liferay.com/browse/LPS-168539

@dantewang Rebased from https://github.com/liferay-core-infra/liferay-portal/pull/3149

After implementing the test for [LPS-166969](https://issues.liferay.com/browse/LPS-166969) (see https://github.com/liferay-search/liferay-portal/pull/626) I have realized that we are repeating the BundleUtil.getBundle method code everywhere.

In this PR I am replacing all the repeated code with a direct call to BundleUtil.getBundle

I am also moving the existing BundleUtil class from portal-impl to portal-kernel as @tinatian requested here: https://github.com/liferay-core-infra/liferay-portal/pull/2451#issuecomment-1398751798

This BundleUtil is used from both portal-impl classes and OSGI modules code so I think the best option is to move there.

### CODE REVIEW

**Comment by catarineaguiar** [comment]
no regressions found.

**Comment by dantewang** [comment]
@tinatian This pull moves the util to portal-kernel (the reasonable place) first, as you commented in https://github.com/liferay-core-infra/liferay-portal/pull/2451#issuecomment-1398751798; however, I'm not very sure about the "centralization" and issue with `BundleUtil` itself in your comment. Could you take a look?

**Comment by jorgediaz-lr** [comment]
Hi @dantewang @tinatian 

Is there any update about this pull request?

Thank you

**Comment by jorgediaz-lr** [comment]
Hi @tinatian @dantewang 

I have rebased this PR to avoid a code conflict.

Regards,
Jorge

File: modules/apps/asset/asset-publisher-test/src/testIntegration/java/com/liferay/asset/publisher/upgrade/v1_0_0/test/UpgradePortletPreferencesTest.java [hunk 1]
```diff
@@ -407,24 +406,8 @@ protected void setUpUpgradePortletPreferences()
...
-				assetPublisherWebBundle = curBundle;
-
-				break;
-			}
-		}
-
-		Assert.assertNotNull(
-			"Unable to find asset-publisher-web bundle",
-			assetPublisherWebBundle);
+		Bundle assetPublisherWebBundle = BundleUtil.getBundle(
```
Comment (tinatian) [inline modules/apps/asset/asset-publisher-test/src/testIntegration/java/com/liferay/asset/publisher/upgrade/v1_0_0/test/UpgradePortletPreferencesTest.java:409]: @jorgediaz-lr You are missing the assert logic after the apply

File: modules/apps/asset/asset-publisher-test/src/testIntegration/java/com/liferay/asset/publisher/web/internal/scheduler/helper/test/AssetEntriesCheckerHelperTest.java [hunk 2]
```diff
@@ -184,24 +184,8 @@ private void _setUpAssetEntriesCheckerHelper() throws Exception {
...
-
-				break;
-			}
-		}
-
-		Assert.assertNotNull(
-			"Unable to find asset-publisher-web bundle",
-			assetPublisherWebBundle);
+		Bundle assetPublisherWebBundle = BundleUtil.getBundle(
+			bundle.getBundleContext(), "com.liferay.asset.publisher.web");
```
Comment (tinatian) [inline modules/apps/asset/asset-publisher-test/src/testIntegration/java/com/liferay/asset/publisher/web/internal/scheduler/helper/test/AssetEntriesCheckerHelperTest.java:188]: @jorgediaz-lr Missing assert again

File: modules/apps/configuration-admin/configuration-admin-test/src/testIntegration/java/com/liferay/configuration/admin/internal/util/test/ConfigurationDDMFormDeclarationUtilTest.java [hunk 3]
```diff
@@ -61,23 +60,8 @@ public void setUp() throws Exception {
...
-
-				break;
-			}
-		}
-
-		if (bundle == null) {
-			throw new IllegalStateException(
-				"com.liferay.configuration.admin.web bundle not found");
-		}
+		Bundle bundle = BundleUtil.getBundle(
```
Comment (tinatian) [inline modules/apps/configuration-admin/configuration-admin-test/src/testIntegration/java/com/liferay/configuration/admin/internal/util/test/ConfigurationDDMFormDeclarationUtilTest.java:63]: @jorgediaz-lr You are missing the IllegalStateException throwing logic

**Comment by tinatian** [comment]
@jorgediaz-lr 
Please see my comments, please review the changes in https://github.com/liferay-core-infra/liferay-portal/pull/3332/commits/a9fb0ec86d45853eed330ab0c7206714909b01e8, it should not change the logic while applying a new API method. I did not check all of them, please make sure you review all the changes.

Please resend.

Thanks.
Tina

**Comment by jorgediaz-lr** [comment]
Hi @tinatian I removed the Assert.assertNotNull and the IllegalStateException throwing logic because the `BundleUtil.getBundle` method was throwing an IllegalStateException instead of a null pointer when the bundle didn't exist, that validation weren't necessary anymore.

I have sent a new PR that avoids changing that code and maintains the same logic everywhere see: https://github.com/liferay-core-infra/liferay-portal/pull/4247

**Comment by jorgediaz-lr** [comment]
Hi @tinatian I removed the _Assert.assertNotNull_ and the _IllegalStateException_ throwing logic because the `BundleUtil.getBundle` method was throwing an IllegalStateException instead of a null pointer when the bundle didn't exist, so that validation weren't necessary anymore.

I have sent a new PR that avoids changing that code and maintains the same logic everywhere see: https://github.com/liferay-core-infra/liferay-portal/pull/4247

**Pull request closed without merging**